docs: document cookie-group-collect app script hook (#2452)
shopware/docs: guides/plugins/apps/storefront/cookies-with-apps.md

</manifest>
```

## Showing cookies conditionally

> Cookies declared in the `manifest.xml` are shown on every sales channel. If your cookies are only relevant under certain conditions, for example a payment provider cookie that is only needed when the corresponding payment method is active in the current sales channel, you can remove them again with an app script for the `cookie-group-collect` hook.

Reviewer: ```suggestion
Cookies declared in `manifest.xml` are shown across all sales channels. If your cookies are only relevant under certain conditions, for example, a payment provider cookie that is only needed when the corresponding payment method is active in the current sales channel, you can remove them again with an app script for the `cookie-group-collect` hook.
```

The hook is triggered whenever the cookie consent groups are collected. It gives you access to the collected cookie groups of all extensions, so your script can remove your own groups and entries based on any condition you can express in a script:

```twig
{# Resources/scripts/cookie-group-collect/filter-payment-cookies.twig #}
{% set criteria = {
    'filter': [
        { 'type': 'equals', 'field': 'active', 'value': true },
        { 'type': 'equals', 'field': 'handlerIdentifier', 'value': 'app\\MyApp_myPaymentMethod' }
    ]
} %}

{% if services.store.search('payment_method', criteria).total == 0 %}
    {% set group = hook.cookieGroups.get('myapp.cookie-group.name') %}

    {% if group is not null and group.entries is not null %}
        {% do group.entries.remove('myapp_payment_cookie') %}
    {% endif %}
{% endif %}
```

> Since the script runs after all cookies were collected, the entries declared in your `manifest.xml` are already present and can be removed by their cookie name. Groups are indexed by their `snippet-name`, entries by their cookie name.

Reviewer: ```suggestion
Since the script runs after all cookies have been collected, the entries declared in your `manifest.xml` are already present and can be removed by their cookie names. Groups are indexed by their `snippet-name`, entries by their cookie name.
```
